bindings_swift/README.md

```diff
@@ -39,3 +39,103 @@ It pairs with [xmtp-rust-swift](https://github.com/xmtp/xmtp-rust-swift) which i
+- Make sure you're in `bindings_swift` crate
+- After you make your code changes (and maybe even add a unit test) the utter: `make swift` to build and push the XMTPRustSwift.xcframework file to xmtp-rust-swift
+
+Possible issues: You need toolchains (run `make download-toolchains` from above)
+
+3. **Go into `xmtp-rust-swift` and check that there are diffs `git diff`**
+- You should see a changeset similar but not identical to:
+  ```
+    modified:   Sources/XMTPRust/xmtp_rust_swift.swift
+    modified:   XMTPRustSwift.xcframework/Info.plist
+    modified:   XMTPRustSwift.xcframework/ios-arm64/Headers/Generated/xmtp_rust_swift/xmtp_rust_swift.h
+    modified:   XMTPRustSwift.xcframework/ios-arm64/libxmtp_rust_swift.a
+    modified:   XMTPRustSwift.xcframework/ios-arm64_x86_64-simulator/Headers/Generated/xmtp_rust_swift/xmtp_rust_swift.h
+    modified:   XMTPRustSwift.xcframework/ios-arm64_x86_64-simulator/libxmtp_rust_swift.a
+    modified:   XMTPRustSwift.xcframework/macos-arm64_x86_64/Headers/Generated/xmtp_rust_swift/xmtp_rust_swift.h
+    modified:   XMTPRustSwift.xcframework/macos-arm64_x86_64/libxmtp_rust_swift.a
+    modified:   Sources/XMTPRust/xmtp_rust_swift.swift
+  ```
+- Check that XMTPRustSwift.xcframework is roughly ~50-60MB, if not then the `bindings_swift` build profile is messed up and not optimizing built binary size correctly. Check the Cargo.toml and check the top-level Cargo.toml workspace for libxmtp.
+```
+$ du -hs XMTPRustSwift.xcframework
+55M    XMTPRustSwift.xcframework
+```
+- Create a new local branch like `git checkout -b my_local_hello_from_rust`
+
+4. **Now switch repos again to `xmtp-ios`**
+- Follow README steps in xmtp-ios to get set up
+- Open xmtp-ios in Xcode as a directory. Do not open the `example/*.xcodeproject` file as the entrypoint
+- You should see Swift packages syncing on the left side
+- Open up `Package.swift` and replace the xmtp-rust-swift branch (see a few sections above for xmtp-ios integration)
+- Run the unit tests in Xcode, at this point the steps are standard to xmtp-ios (not Rust-integration specific) and documented in the xmtp-ios README.md
+- Find a location in the code that uses `XMTPRust`'s sha256 or keccak256 functions. Add an additional call to `hello_from_rust`.
+- Test that the call compiles correctly and emits the correct string
+
+5. **OPTIONAL: You want to push a Cocoapod**
+- There are two cocoapods in play. Cocoapod `XMTPRust` comes from the `xmtp-rust-swift` repo and `XMTP` comes from `xmtp-ios` and depends on `XMTPRust`
+- Make sure your local branch of xmtp-rust-swift passes `pod lib lint XMTPRust.podspec --allow-warnings`
+- To push the `XMTPRust` cocoapod, go to `xmtp-rust-swift` and bump the version in the `XMTPRust.podspec` file to your intended one
+- Open a PR, get it merged (push branches remotely with care, as the framework files form a 40MB diff, I only do it when I'm very confident in local changes)
```

Comment:
Those are some big diffs. Is `git-lfs` not an option?

Reply:
Haven't looked into git-lfs but definitely worth checking out! As long as the Cocoapod clone handles lfs we're good

Reply:
My guess is that it'll work as long as consumers have `git-lfs` installed on their systems: https://stackoverflow.com/q/64438117